torch/csrc/inductor/aoti_runtime/arrayref_tensor.h

```diff
@@ -215,7 +215,7 @@ class ArrayRefTensor {
         &result));
     void* dataPtr = nullptr;
     AOTI_TORCH_ERROR_CODE_CHECK(aoti_torch_get_data_ptr(result, &dataPtr));
-    std::memcpy(dataPtr, data(), numel() * sizeof(T));
+    std::memcpy(dataPtr, data(), strides_.data()[strides_.size() - 1] * numel() * sizeof(T));
```

Comment:
noob question: curious if this is the a good long term fix. It seems it could be expensive in some cases?

Reply:
I think it should be fine because it's in the "expensive" copy function :)

Reply:
hmm looks like this change fixed the `fft` test, but broke `test_dynamic_scalar` and `test_return_view_constant`. I'm looking into it. Meanwhile do you have any insight about why this might cause the two tests to fail? Thanks!

Reply:
oh ok, I think they failed and segfaults because `_strides.size()` is 0. I'll just add an if-else on stride size.

Reply:
@chenyang78 @henrylhtsang  oh I figured out what's wrong with `test_return_view_constant`, but not sure about how to fix it yet. In this case, the stride is `[1 5]` because we did a transpose, so we can't just multiply by the last dim.  Do you think something like `max_stride_number * numel() * sizeof(T)` would work?

Reply:
I think there might be something wrong with how we create the MiniArrayRef. Sometimes when we call  MiniArrayRef(T* data, size_t length), `data` contains more elements than just `length`. We should fix it there, instead of letting `expensiveCopyToTensor()` to infer how much elements `MiniArrayRef` has. WDTY?

Reply:
> @chenyang78 @henrylhtsang oh I figured out what's wrong with `test_return_view_constant`, but not sure about how to fix it yet. In this case, the stride is `[1 5]`, because we did a transpose, so we can't just multiply by the last dim. The correct computation is ` 1 * numel() * sizeof(T)`. I'm not sure how do we know the real total size of the underlying tensor?

Yeah, I don't think the current implementation is fully functional. Col-major is a simple case that it doesn't handle correctly. There are more unsupported cases. We might want to borrow some idea from ```clone_preserve_strides```:

https://github.com/pytorch/pytorch/blob/2ee989530467eeabcd8beee741262d042ebea3e4/aten/src/ATen/native/TensorShape.cpp#L3976

Reply:
I think the problem with `ArrayRef` and `MiniArrayRef` now is that this true number of underlaying elements is not stored anywhere (something equivalent of `self.storage().sym_nbytes()` in `clone_preserve_strides`). Maybe we should add a `storage_size` attribute to `MiniArrayRef` so we know the underlaying size of the data. WDYT?

Reply:
I pushed a change to use the numel from `clone_preserve_strides` instead of the default numel. 

https://github.com/pytorch/pytorch/blob/2ee989530467eeabcd8beee741262d042ebea3e4/aten/src/ATen/native/TensorShape.cpp#L3976

Reply:
Very nice solution! Thanks.